server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java

```diff
@@ -1174,6 +1174,35 @@ public TSExecuteStatementResp insert(TSInsertionReq req) {
     }
   }
 
+  @Override
+  public TSStatus insertRowInBatch(TSInsertInBatchReq req) {
+    if (!checkLogin()) {
+      logger.info(INFO_NOT_LOGIN, IoTDBConstant.GLOBAL_DB_NAME);
+      return new TSStatus(getStatus(TSStatusCode.NOT_LOGIN_ERROR));
+    }
+
+    TSStatus status = null;
+    for (int i = 0; i < req.deviceIds.size(); i++) {
+      InsertPlan plan = new InsertPlan();
+      plan.setDeviceId(req.getDeviceIds().get(i));
+      plan.setTime(req.getTimestamps().get(i));
+      plan.setMeasurements(req.getMeasurementsList().get(i).toArray(new String[0]));
+      plan.setValues(req.getValuesList().get(i).toArray(new String[0]));
+      status = checkAuthority(plan);
+      if (status != null) {
+        return new TSStatus(status);
+      }
+
+      status = executePlan(plan);
+      if (status.getStatusType().getCode() != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        return status;
+      }
+    }
+
+    return status;
```

Comment:
this can be a list, just like jdbc execute batch statements

Reply:
fixed~